### ISSUE TITLE
[Grid] expanding a detail row causes unexpected behavior

### ISSUE THREAD

**Issue by svetq** [telerik/kendo-angular#2297]
**Describe the bug**
When setting [navigable]="true" and using a master-detail grid with virtual scrolling, if we scroll down a bit and try to expand any of the detail rows then the Grid will be scrolled to a different location.

**To Reproduce**
https://stackblitz.com/edit/angular-21gqva?file=app/app.component.ts

1. Scroll down.
2. Expand any detail row

**Expected behavior**
The scroll position should not change when expanding detail rows.

**Comment by dtopalov** [comment]
Also discussed in a private support thread where the customer provided additional details and reported further problems that seem related (observable in the following Stackblitz demo):

[https://stackblitz.com/edit/angular-hlju9w](https://stackblitz.com/edit/angular-hlju9w)

"I think although here it says that the issue occurred when they attempt to open expand the row, and we said that it occurs when we attempt to click and focus a cell in the row, it actually happens in both cases.

So, on further poking of this test case and ours, it seems like maybe there are three separate but related issues:

1. Clicking on a row past the fifth or so doesn't do whatever it would normally do (focus the cell, expand the row); instead, it jumps down several rows. Sometimes it focuses/expands whatever row jumps into view (i.e. I clicked row 5 but it jumps down to row 25 and expands that).

To reproduce, scroll down past the 10th row and click on a cell.

1. The keyboard navigation skips rows when I use the up/down arrows. (Possibly because it's giving focus to the not-currently-visible detail row?) If I attempt to press left or right arrows when the focus is in-between rows, I get into a state where the keyboard navigation no longer works at all.

1. When I've got a grid that is editable and master-detail, focusing the expand/collapse cell and pressing Enter gives focus to the first editable cell and activates its edit mode.
